### ISSUE TITLE
Expected input batch_size to match target batch_size

### ISSUE THREAD

**Issue by amoelle** [deepset-ai/FARM#100]
Hello,

when running the Trainer for a multi-class classification (4 classes, batch_size=8) there seems to be a missmatch:

`Expected input batch_size (8) to match target batch_size (32)`

The error appears in this  context:

    farm/modeling/prediction_head.py(262)logits_to_loss()
    -> return self.loss_fct(logits, label_ids.view(-1))
    (Pdb)  logits
    tensor([[ 0.0025,  0.1011,  0.0897,  0.5420],
            [-0.2871,  0.1540,  0.1067,  0.3153],
            [ 0.0552,  0.5559,  0.0795,  0.3348],
            [ 0.2319,  0.1192, -0.0756,  0.5185],
            [ 0.2457, -0.1189, -0.3208,  0.1301],
            [-0.1660,  0.1726, -0.0196,  0.0887],
            [-0.3584,  0.0351,  0.0103,  0.2665],
            [-0.3831,  0.0922,  0.2931,  0.2717]], grad_fn=<A
    (Pdb)  label_ids
    tensor([[0, 1, 0, 0],
            [0, 0, 1, 0],
            [0, 1, 0, 0],
            [0, 0, 0, 1],
            [1, 0, 0, 0],
            [0, 1, 0, 0],
            [1, 0, 0, 0],
            [0, 1, 0, 0]])

.
.
.

    --Call--
    > /usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py(537)__call__()
    -> def __call__(self, *input, **kwargs):
    (Pdb)  input
    (tensor([[ 0.0025,  0.1011,  0.0897,  0.5420],
            [-0.2871,  0.1540,  0.1067,  0.3153],
            [ 0.0552,  0.5559,  0.0795,  0.3348],
            [ 0.2319,  0.1192, -0.0756,  0.5185],
            [ 0.2457, -0.1189, -0.3208,  0.1301],
            [-0.1660,  0.1726, -0.0196,  0.0887],
            [-0.3584,  0.0351,  0.0103,  0.2665],
            [-0.3831,  0.0922,  0.2931,  0.2717]], grad_fn=<AddmmBackward>), tensor([0, 1, 0, 
            0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0,
            1, 0, 0, 0, 0, 1, 0, 0]))

Then input and target missmatch:

    --Call--
    > /usr/local/lib/python3.7/dist-packages/torch/nn/modules/loss.py(914)forward()
    -> def forward(self, input, target):
    (Pdb)  input
    tensor([[ 0.0025,  0.1011,  0.0897,  0.5420],
        [-0.2871,  0.1540,  0.1067,  0.3153],
        [ 0.0552,  0.5559,  0.0795,  0.3348],
        [ 0.2319,  0.1192, -0.0756,  0.5185],
        [ 0.2457, -0.1189, -0.3208,  0.1301],
        [-0.1660,  0.1726, -0.0196,  0.0887],
        [-0.3584,  0.0351,  0.0103,  0.2665],
        [-0.3831,  0.0922,  0.2931,  0.2717]], grad_fn=<AddmmBackward>)
    (Pdb)  target
    tensor([0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0,
        1, 0, 0, 0, 0, 1, 0, 0])

Finally the error is thrown here:

    > /usr/local/lib/python3.7/dist-packages/torch/nn/functional.py(1820)nll_loss()
    -> if input.size(0) != target.size(0):
    -> raise ValueError('Expected input batch_size ({}) to match target batch_size ({}).'

Is there something i can do to avoid this Value Error?

**Comment by amoelle** [comment]
Hi @Timoeller, thank you very much for the support!
I have changed the 'multilabel' parameter of the TextClassificationProcessor from True to False now and the issue seems to be resolved (miss-interpreted the parameter, sorry for the inconvenience! 🙈)
